File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let output = [];
        for (let i = 0; i < nums.length; i++) {
            let valor_i = nums[i];
            nums[i] = 1;
            let output_i = nums.reduce((acumulador, valorAtual) => acumulador * valorAtual, 1);
            output.push(output_i);
            nums[i] = valor_i;
        }
        return output
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        output = nums[0]
        curSum = 0

        for num in nums:
            if curSum < 0:
                curSum = 0
            
            curSum += num
            output = max(output, curSum)

        return output

        